custom/abdm/models.py

```diff
@@ -0,0 +1,32 @@
+import os
+import binascii
+
+from django.db import models
+
+
+class ABDMUser(models.Model):
+    username = models.CharField(max_length=100, primary_key=True)
+    access_token = models.CharField(max_length=2000, null=True, blank=True)
+    domain = models.CharField(max_length=100, default="")
+
+    def save(self, *args, **kwargs):
+        if not self.access_token:
+            self.generate_token()
+        return super().save(*args, **kwargs)
+
+    def generate_token(self):
+        token = binascii.hexlify(os.urandom(20)).decode()
+        self.access_token = token
+
+    @property
+    def is_token_valid(self):
+        # To be used in future when token expiry is introduced.
+        return True
+
+    @property
+    def is_authenticated(self):
+        """
+        Always return True. This is a way to tell if the user has been
+        authenticated in REST views.
+        """
+        return True
```

Comment:
this does not look right. Can you share why this is okay to do or why this is needed on the user model and where it would be called?

Reply:
Since we are using custom class to authenticate user, this is always returning `True` as DRF will be calling this method during authentication. Actual auth part takes place in `ABDMUserAuthentication` class.

Reply:
So we are just placing it because it would be called internally? We are placing the decorator on the view, so if we don't do that, this won't be called?

Reply:
> We are placing the decorator on the view, so if we don't do that, this won't be called?

Yes, that's my understanding. Ref. https://github.com/dimagi/commcare-hq/pull/32332#discussion_r1053766743

Reply:
Ya, so why are we decorating the view with it if its not really checking anything?